### ISSUE TITLE
In GTK3 gtk_menu_popup is an error

### ISSUE THREAD

**Issue by MAOmao000** [wxWidgets/wxWidgets#22328]
ubuntu 20.04
wxWidgets 3.1.6
gtk libgtk-3-0

The gtk_menu_popup function on line 5897 of wxWidgets-3.1.6 /src/gtk/window.cpp uses the gtk_menu_popup function.
On the other hand, in the GTK3 documentation,
  Note that this function does not work well on GDK backends that do not have global coordinates such as Wayland or Mir.
  You probably need to use one of the gtk_menu_popup_at_variants that doesn't have this problem.
  It is deprecated since version 3.22 and should not be used in newly written code.
It is described as.
When actually operating it, the following warning message is output on GTK3.
  Gdk-Message: 13: 41: 10.110: Window 0x55dbdb0012c0 is a temporary window without parent,
                             application will not be able to position it on screen.
  Gdk-CRITICAL **: 13: 41: 10.111: gdk_wayland_window_handle_configure_popup: assertion'impl-> transient_for' failed
If you change gtk_menu_popup in line 5897 of wxWidgets-3.1.6 /src/gtk/window.cpp to gtk_menu_popup_at_pointer, the warning message will disappear.
Would you please deal with it?
Thank you.

**Comment by vadz** [comment]
I don't think it's a good idea to unconditionally ignore the position and show the menu at the mouse position. It looks like we ought to use `gtk_menu_popup_at_widget()` or `gtk_menu_popup_at_rect()` instead.

BTW, what is the window `0x55dbdb0012c0` in your case? As usual, having a way to reproduce the problem would be very helpful. Do you see it it in the menu wxWidgets sample, which uses `PopupMenu()`? If not, can you please make a [small patch](https://www.wxwidgets.org/develop/how-to-submit-patches/) showing it there? TIA!

**Comment by MAOmao000** [comment]
Thank you for your support.
The reproduction procedure is as follows.
```
~/wxWidgets-3.1.6$ ./configure --disable-shared --enable-unicode --with-sdl
Configured wxWidgets 3.1.6 for `x86_64-pc-linux-gnu'

  Which GUI toolkit should wxWidgets use?                 GTK+ 3 with support for GTK+ printing libnotify
  Should wxWidgets be compiled into single library?       no
  Should wxWidgets be linked as a shared library?         no
  Should wxWidgets support Unicode?                       yes (using wchar_t)
  What level of wxWidgets compatibility should be enabled?
                                       wxWidgets 2.8      no
                                       wxWidgets 3.0      yes
  Which libraries should wxWidgets use?
                                       STL                no
                                       jpeg               builtin
                                       png                sys
                                       regex              sys
                                       tiff               builtin
                                       lzma               no
                                       zlib               sys
                                       expat              sys
                                       libmspack          no
                                       sdl                yes

~/wxWidgets-3.1.6$ make
~/wxWidgets-3.1.6$ sudo dpkg -r wxwidgets
~/wxWidgets-3.1.6$ sudo checkinstall
~/wxWidgets-3.1.6$ ls -la /usr/local/lib
total 68748
drwxr-xr-x  7 root root      4096 Apr 19 00:09 .
drwxr-xr-x 13 root root      4096 Apr 19 00:08 ..
drwxr-xr-x  3 root root      4096 Apr 14 11:08 cmake
-rw-r--r--  1 root root    798794 Apr 16 08:42 libSDL2_test.a
-rw-r--r--  1 root root      6372 Apr 16 08:42 libSDL2main.a
-rw-r--r--  1 root root   8117038 Apr 19 00:08 libwx_baseu-3.1.a
-rw-r--r--  1 root root   1236334 Apr 19 00:08 libwx_baseu_net-3.1.a
-rw-r--r--  1 root root    135492 Apr 19 00:08 libwx_baseu_xml-3.1.a
-rw-r--r--  1 root root      1212 Apr 19 00:08 libwx_gtk3u_adv-3.1.a
-rw-r--r--  1 root root   2064904 Apr 19 00:08 libwx_gtk3u_aui-3.1.a
-rw-r--r--  1 root root  33191134 Apr 19 00:08 libwx_gtk3u_core-3.1.a
-rw-r--r--  1 root root    284188 Apr 19 00:08 libwx_gtk3u_gl-3.1.a
-rw-r--r--  1 root root   2732272 Apr 19 00:08 libwx_gtk3u_html-3.1.a
-rw-r--r--  1 root root   2760700 Apr 19 00:08 libwx_gtk3u_propgrid-3.1.a
-rw-r--r--  1 root root    413116 Apr 19 00:08 libwx_gtk3u_qa-3.1.a
-rw-r--r--  1 root root   1676342 Apr 19 00:08 libwx_gtk3u_ribbon-3.1.a
-rw-r--r--  1 root root   4951638 Apr 19 00:08 libwx_gtk3u_richtext-3.1.a
-rw-r--r--  1 root root   1154532 Apr 19 00:08 libwx_gtk3u_stc-3.1.a
-rw-r--r--  1 root root   5492790 Apr 19 00:08 libwx_gtk3u_xrc-3.1.a
-rw-r--r--  1 root root    341846 Apr 19 00:08 libwxjpeg-3.1.a
-rw-r--r--  1 root root   4296390 Apr 19 00:08 libwxscintilla-3.1.a
-rw-r--r--  1 root root    674810 Apr 19 00:08 libwxtiff-3.1.a
drwxr-xr-x  2 root root      4096 Apr 16 08:43 pkgconfig
drwxrwsr-x  4 root staff     4096 Jul 23  2021 python2.7
drwxrwsr-x  3 root staff     4096 Jun  4  2021 python3.8
drwxr-xr-x  4 root root      4096 Apr 19 00:08 wx
~$ cd ../
~$ rm -rf LeelaGUI
~$ git clone https://github.com/MAOmao000/LeelaGUI.git
~$ cd LeelaGUI
~/LeelaGUI$ make
~/LeelaGUI$ ./leelagui
Gdk-Message: 00:39:05.529: Window 0x5597c02b9ba0 is a temporary window without parent, application will not be able to position it on screen.
(leelagui:1937): Gdk-CRITICAL **: 00:39:05.529: gdk_wayland_window_handle_configure_popup: assertion 'impl->transient_for' failed
```
The place to actually call PopupMenu is as follows.
In response to the event from the grid control, the pop-up menu will be displayed below.
AnalysisWimdow.cpp (line 214)
```
void AnalysisWindow::doContextMenu(wxGridEvent& event) {
    size_t row = event.GetRow();
    wxMenu mnu;
    mnu.SetClientData((void*)row);
    mnu.Append(ID_COPYPV, _("Copy PV"));
    mnu.Append(ID_COPYLINE, _("Copy entire line"));
    mnu.AppendSeparator();
    mnu.Append(ID_DESELECTLINE, _("Deselect line"));
    mnu.Bind(wxEVT_COMMAND_MENU_SELECTED, &AnalysisWindow::onContextMenuClick, this);
    PopupMenu(&mnu);
}
```
![image](https://user-images.githubusercontent.com/49147822/163837432-d6742665-279c-4b5b-9818-5e97cd0178b4.png)
![image](https://user-images.githubusercontent.com/49147822/163837538-d2cbb84c-14f8-4a87-983c-85770797d8b5.png)
I tried wxWidgets-3.1.6 sample samples / widgets but didn't get the same result.
```
~/wxWidgets-3.1.6$ cd samples/widgets
~/wxWidgets-3.1.6/samples/widgets$ make
~/wxWidgets-3.1.6/samples/widgets$ ./widgets
(widgets:32713): Gdk-WARNING **: 00:13:56.255: Tried to map a popup with a non-top most parent
(widgets:32713): Gdk-WARNING **: 00:14:51.334: Tried to map a popup with a non-top most parent
(widgets:32713): Gdk-WARNING **: 00:14:59.150: Tried to map a popup with a non-top most parent
```
![image](https://user-images.githubusercontent.com/49147822/163838154-8e8db807-4e1b-43ce-bf9d-40b6fde081a6.png)
![image](https://user-images.githubusercontent.com/49147822/163838215-29c1803b-b3bf-4497-8056-ef6a0ca7a5ca.png)

**Comment by vadz** [comment]
Hmm, for a `PopupMenu()` call with the default `pos` value we probably could indeed just use `gtk_menu_popup_at_pointer()` which would be even simpler than using `gtk_menu_popup_at_rect()`.

Any PRs doing this (with the required compile- and run-time version checks) would be welcome, but if not I'll try to do it myself later.

**Comment by paulcor** [comment]
Fixed via a261c80298666fe6f97c75553f39a0e7ec23bf6c
